### ISSUE TITLE
Update ios sdk to version 12.1.0

### ISSUE THREAD

**Issue by thisisthekap** [xamarin/FacebookComponents#223]
The iOS bindings need to be updated to support new ad measurement features which were mandatory after the introduction of iOS 14.5.

@SotoiGhost As @dalexsoto informed me, you are the original maintainer of this repository. If needed, I am happy to help with doing the update, as we need the update rather sooner than later.

@SotoiGhost Please let me know how we best proceed on this one.

**Comment by SotoiGhost** [comment]
@thisisthekap Thanks for the heads up. Please, feel free to do the update if possible, that would be a great help for us! I have this update in my TODO list, but I'm attending another task before getting into this one. Let me know if you decide to do the update. Thank you in advance!

**Comment by thisisthekap** [comment]
@SotoiGhost When do you think you would be going to do the upgrade?

Unfortunately, I got a lot on my plate as well as of now.

**Comment by Huaba93** [comment]
@SotoiGhost We also need the update. It is important so that we can use aggregated event measurement for apps. But unfortunately I don't know how to do the binding update

To configure Aggregated Event Measurement for your app:
Update to Facebook SDK for iOS version 11.2.0 or above to configure your app events._

Source: https://www.facebook.com/business/help/2750680505215705?id=428636648170202

**Comment by SotoiGhost** [comment]
I'll start doing the update for this

**Comment by Huaba93** [comment]
@SotoiGhost thank you very much 🙏.

ETA? The Christmas sales are just around the corner. 🙈

**Comment by Huaba93** [comment]
@SotoiGhost any news? or is there anything I can help you with? eg. testing?

**Comment by thisisthekap** [comment]
@SotoiGhost Any news on your timeline?

**Comment by MateuszLas421** [comment]
Any news ?

**Comment by thisisthekap** [comment]
@MateuszLas421 @SotoiGhost @Huaba93 I started working on the bindings myself today.

**Comment by SotoiGhost** [comment]
Sorry folks for the long waiting on this folks. I almost have a working PR for this. I'm hoping to post it by today.

**Comment by thisisthekap** [comment]
@SotoiGhost Thanks for that. I'll stop working on that now. I would appreciate better communication on planning and ETAs the next time.

**Comment by thisisthekap** [comment]
@SotoiGhost Do you need some assistance e.g. in testing the updated version?

**Comment by tigreye007** [comment]
I tried to update to 12.2.0 today and received the following error message:

Severity	Code	Description	Project	File	Line	Suppression State
Error		Could not find a part of the path 'C:\Users\tigre\.nuget\packages\xamarin.facebook.sharekit.ios\12.2.0\lib\xamarinios10\Facebook.ShareKit.resources\FBSDKShareKit.xcframework\ios-arm64_i386_x86_64-simulator\FBSDKShareKit.framework\Modules\FBSDKShareKit.swiftmodule\arm64-apple-ios-simulator.swiftdoc'.				

Any ideas?

**Comment by thisisthekap** [comment]
@tigreye007 Do you experience this issue when building your project?

**Comment by tigreye007** [comment]
> @tigreye007 Do you experience this issue when building your project?

I got this when I tried to upgrade the package via nuget (from 9.0.1.5 => 12.2.0)

**Comment by tigreye007** [comment]
My guess is that it doesn't like the path length (263 characters)

**Comment by AlexanderTretyackov** [comment]
> I tried to update to 12.2.0 today and received the following error message:
> 
> Severity Code Description Project File Line Suppression State Error Could not find a part of the path 'C:\Users\tigre.nuget\packages\xamarin.facebook.sharekit.ios\12.2.0\lib\xamarinios10\Facebook.ShareKit.resources\FBSDKShareKit.xcframework\ios-arm64_i386_x86_64-simulator\FBSDKShareKit.framework\Modules\FBSDKShareKit.swiftmodule\arm64-apple-ios-simulator.swiftdoc'.
> 
> Any ideas?

Same problem

**Comment by VanjaPopovic** [comment]
Same issue here

**Comment by denisminb** [comment]
Same problem

"C:\Users\HomeIT\.nuget\packages\xamarin.facebook.sharekit.ios\12.2.0\lib\xamarinios10\Facebook.ShareKit.resources\FBSDKShareKit.xcframework\ios-arm64_i386_x86_64-simulator\FBSDKShareKit.framework\Modules\FBSDKShareKit.swiftmodule\arm64-apple-ios-simulator.swiftdoc".

**Comment by tigreye007** [comment]
Has anyone found a workaround for this yet?

**Comment by AlexanderTretyackov** [comment]
> Has anyone found a workaround for this yet?

I work on windows 10, so there is an way to [enable long paths,](https://docs.microsoft.com/ru-ru/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later), but anyway I can't build my project with that nuget because of such error:
`/usr/bin/ditto C:/n/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/ bin/iPhone/Release/[BioGenom.iOS.app/../FBSDKCoreKit.framework.dSYM](http://biogenom.ios.app/FBSDKCoreKit.framework.dSYM)  (Идентификатор задачи TaskId:239)
2>  ditto: Cannot get the real path for source 'C:/n/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/' (Идентификатор задачи TaskId:239)
2>C:\Program Files (x86)\Microsoft Visual Studio\2019_Latest\MSBuild\Xamarin\iOS\Xamarin.Shared.targets(2050,3): error MSB6006: "ditto" завершилась с кодом 1.`

**Comment by denisminb** [comment]
> Has anyone found a workaround for this yet?

no, not solved yet

**Comment by denisminb** [comment]
> > Has anyone found a workaround for this yet?
> 
> I work on windows 10, so there is a way to [enable long paths,](https://docs.microsoft.com/ru-ru/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later), but anyway I can't build my project with that nuget because of such error: `/usr/bin/ditto C:/n/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/ bin/iPhone/Release/[BioGenom.iOS.app/../FBSDKCoreKit.framework.dSYM](http://biogenom.ios.app/FBSDKCoreKit.framework.dSYM) (Идентификатор задачи TaskId:239) 2> ditto: Cannot get the real path for source 'C:/n/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/' (Идентификатор задачи TaskId:239) 2>C:\Program Files (x86)\Microsoft Visual Studio\2019_Latest\MSBuild\Xamarin\iOS\Xamarin.Shared.targets(2050,3): error MSB6006: "ditto" завершилась с кодом 1.`

I also tried to include long paths in win10, manually create the required folder, but nothing helps

**Comment by arscoloramanfredi** [comment]
Same issue here, solutions? We are stuck with 9.0.1.5 but it's not working anymore.

**Comment by SotoiGhost** [comment]
Folks, does this happen using Visual Studio or also with command line?

If this is happening with Visual Studio, please, clear your NuGet cache, delete the `bin` and `obj` folder and do a restore from command line before opening the solution with VS. Let me know if this works for you.

**Comment by denisminb** [comment]
> Folks, does this happen using Visual Studio or also with command line?
> 
> If this is happening with Visual Studio, please, clear your NuGet cache, delete the `bin` and `obj` folder and do a restore from command line before opening the solution with VS. Let me know if this works for you.

I clear your NuGet cache, deleted the the bin and obj folder, restored the packages via nuget restore 'path my project' , before opening the project from visual studio, then opened the project and tried to update the package, but the error message is still the same:
`C:\Users\HomeIT\.nuget\packages\xamarin.facebook.sharekit.ios\12.2.0\lib\xamarinios10\Facebook.ShareKit.resources\FBSDKShareKit.xcframework\ios-arm64_i386_x86_64-simulator\FBSDKShareKit.framework\Modules\FBSDKShareKit.swiftmodule\arm64-apple-ios-simulator.swiftdoc`

**Comment by SotoiGhost** [comment]
@denisminb Try updating your .csproj or your packages.config, setting to the most recent version of FB package by hand and then do the restore before opening VS.

**Comment by denisminb** [comment]
> @denisminb Try updating your .csproj or your packages.config, setting to the most recent version of FB package by hand and then do the restore before opening VS.

I understand you correctly, manually edit the file .csproj or your packages.config, change the version of the iOS packages from 9.0.1.5 to 12.2.0 and then restore?

**Comment by SotoiGhost** [comment]
@denisminb Yep, and don't forget to clean your NuGet cache, I think deleting the `bin` and `obj` is optional, but just in case.

**Comment by denisminb** [comment]
The package update seems to have been successful, it remains only to check.
![image](https://user-images.githubusercontent.com/45455600/155411776-7c3e3daf-4eed-4e77-a407-6721eb875a7a.png)

**Comment by denisminb** [comment]
> @denisminb Yep, and don't forget to clean your NuGet cache, I think deleting the `bin` and `obj` is optional, but just in case.

The package update seems to have been successful, it remains only to check. Thanks
![image](https://user-images.githubusercontent.com/45455600/155412159-09127573-af6f-455d-941d-d4a2db30847b.png)

**Comment by SotoiGhost** [comment]
@denisminb Thanks for letting me know that this is working, I'll file a bug of this, as this is working correctly using command line

**Comment by arscoloramanfredi** [comment]
In my case, 12.2.0 dependency is added correctly with the method above but compilation is not working. 

ditto: Cannot get the real path for source 'C:/Users/user/.nuget/packages/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/

**Comment by AlexanderTretyackov** [comment]
> In my case, 12.2.0 dependency is added correctly with the method above but compilation is not working.
> 
> ditto: Cannot get the real path for source 'C:/Users/user/.nuget/packages/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/

I have same issue

**Comment by arscoloramanfredi** [comment]
From Visual Studio for Mac 2019 it's working

**Comment by SotoiGhost** [comment]
@arscoloramanfredi @AlexanderTretyackov Can you verify if the dSYM is there?

**Comment by denisminb** [comment]
> @arscoloramanfredi @AlexanderTretyackov Can you verify if the dSYM is there?

I have the same error when building a project in release mode. 
(in debug mode, everything is fine on virtual devices).

```
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Xamarin\iOS\Xamarin.Shared.targets(2069,3): error MSB6006: "ditto" - error number 1
/usr/bin/ditto C:/Users/HomeIT/.nuget/packages/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/ bin/iPhone/Release/Etiket.iOS.app/../FBSDKCoreKit.framework.dSYM 
```

**Comment by arscoloramanfredi** [comment]
> C:/Users/user/.nuget/packages/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/

That folder contains a "Contents" subfolder with plist and DWARF

**Comment by AlexanderTretyackov** [comment]
> > C:/Users/user/.nuget/packages/xamarin.facebook.corekit.ios/12.2.0/buildTransitive/../lib/xamarinios10/Facebook.CoreKit.resources/FBSDKCoreKit.xcframework/ios-arm64_armv7/FBSDKCoreKit.framework/../dSYMs/FBSDKCoreKit.framework.dSYM/
> 
> That folder contains a "Contents" subfolder with plist and DWARF

Same situation

**Comment by denisminb** [comment]
@SotoiGhost Hello. any ideas how to solve the problem?

**Comment by SotoiGhost** [comment]
Can you open a new issue and provide a testcase, please? As I'm not able to repro the issue stated above. Thanks in advance!

**Comment by SotoiGhost** [comment]
I'm seeing the issue now; I already opened an issue of this: https://github.com/xamarin/xamarin-macios/issues/14291

**Comment by SotoiGhost** [comment]
Hi all! This [fix is being reviewed](https://github.com/xamarin/xamarin-macios/pull/14375). Meanwhile, here's a workaround that you could use. Paste it at the bottom of your app `.csproj` before the `</Project>`

```xml
  <Target Name="_WorkaroundForCopyUserFrameworkDebugSymbols"
    Condition="('$(_PlatformName)' == 'macOS') Or ('$(ComputedPlatform)' == 'iPhone' And '$(IsWatchApp)' == 'false')"
    DependsOnTargets="_CompileToNative;_ParseBundlerArguments;"
    BeforeTargets="_CopyUserFrameworkDebugSymbols"
    Inputs="%(_FrameworkNativeReference.Name)"
    Outputs="$(AppBundleDir)\..\%(_FrameworkNativeReference.Filename).framework.dSYM\Contents\Info.plist">

    <ItemGroup>
      <_Source Include="$([System.IO.Path]::GetDirectoryName(%(_FrameworkNativeReference.Name)))\..\dSYMs\%(_FrameworkNativeReference.Filename).framework.dSYM\**\*" />
    </ItemGroup>
    <PropertyGroup>
      <_Destination>$(AppBundleDir)\..\%(_FrameworkNativeReference.Filename).framework.dSYM</_Destination>
    </PropertyGroup>

    <RemoveDir SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Directories="$(_Destination)" />

    <!-- if available copy the provided user frameworks .dSYM -->
    <Ditto
      SessionId="$(BuildSessionId)"
      Condition="'$(IsMacEnabled)' == 'true' And Exists('%(_Source.Identity)')"
      Source="%(_Source.Identity)"
      Destination="$(_Destination)"
    >
    </Ditto>
  </Target>
```
